### PR TITLE
Rayana Alencar backend-python-test

### PR DESCRIPTION
**Pull request by RayanaAlencar**

### CODE REVIEW

File: alayatodo/models.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    username    = db.Column(db.String(255), nullable=False)
+    password    = db.Column(db.String(255), nullable=False)
+    todos       = db.relationship('Todo', backref='user', lazy=True)
+
+    def __repr__(self):
+        return '<User {}>'.format(self.username)
+
+    def set_password(self, password):
+        self.password = generate_password_hash(password)
+
```
Comment (flacours) [inline alayatodo/models.py]: 👍

File: alayatodo/views.py [hunk 2]
```diff
@@ -17,66 +26,129 @@ def home():
...
-        "INSERT INTO todos (user_id, description) VALUES ('%s', '%s')"
-        % (session['user']['id'], request.form.get('description', ''))
-    )
-    g.db.commit()
+    if not (current_user.is_authenticated):
+        return render_template('401.html')
+    user_id = current_user.id
+    description = request.form.get('description', '')
+    new_Todo = Todo(user_id, description,False)
+    try:
```
Comment (flacours) [inline alayatodo/views.py]: scope of try

File: alayatodo/views.py [hunk 3]
```diff
@@ -17,66 +26,129 @@ def home():
...
 @app.route('/todo/<id>', methods=['POST'])
 def todo_delete(id):
-    if not session.get('logged_in'):
-        return redirect('/login')
-    g.db.execute("DELETE FROM todos WHERE id ='%s'" % id)
-    g.db.commit()
+    if not (current_user.is_authenticated):
+        return render_template('401.html')
+    todo = Todo.query.filter_by(id = id).first()
+    try:
```
Comment (flacours) [inline alayatodo/views.py]: what happen when todo is None ?

File: alayatodo/views.py [hunk 4]
```diff
@@ -17,66 +26,129 @@ def home():
...
+@app.route('/todo/<id>/json', methods=['GET'])
+def todo_json(id):
+    if not (current_user.is_authenticated):
+        return render_template('401.html')
+    else:
+        user_id = current_user.id
+        todo = Todo.query.filter_by(id = id).first()
+        if todo is None:
+            return render_template('404.html')
+        if todo.user_id != user_id:
```
Comment (flacours) [inline alayatodo/views.py]: 👍

File: alayatodo/views.py (hunk 3 above)
Comment (RayanaAlencar) [reply alayatodo/views.py]: Hi @flacours I used the postman to see what is the return of this method when todo is None and the return is the todo page with an flash message saying 'Something went wrong. Please try again later'. Thinking about it now, an improvement can be to change it to return a 404 page.

File: alayatodo/views.py (hunk 2 above)
Comment (RayanaAlencar) [reply alayatodo/views.py]: Hi @flacours, can you explain a bit little more ?

**Comment by RayanaAlencar** [comment]
Hi @flacours, 
I really did not think about putting a button to view the Todo as a json, I should have done this, I can update the code. Also, could you go into more detail about the problem of columns in migration?

**Pull request closed without merging**